salt/_modules/metalk8s_package_manager_yum.py

```diff
@@ -143,3 +150,50 @@ def list_pkg_dependents(
             del all_pkgs[pkg_name]
 
     return all_pkgs
+
+
+class MissingPackageException(Exception):
+    '''Missing package exception'''
+    pass
+
+
+def check_pkg_availability(pkgs_info):
+    '''
+    Check that provided packages and their dependencies are available
+
+    pkgs_info
+        Value of pillar key `repo:packages` to consider for the requiring
+        packages to check (format {"<name>": {"version": "<version>"}, ...})
+    '''
+    ybase = yum.YumBase()
+    ybase.conf.tsflags = 'test'
+
+    for name, info in pkgs_info.items():
+
+        version = info.get('version')
+        release = None
+
+        if version == 'None':  # Workaround a bug in jinja templating
+            version = None
+        elif version:
+            version_with_release = version.split('-', 1)
+            if len(version_with_release) == 2:
+                version = version_with_release[0]
+                release = version_with_release[1]
+
+        try:
+            ybase.install(name=name, version=version, release=release)
+        except yum.Errors.InstallError as exc:
+            error = 'No candidate found for package: {0}{1}'.format(
+                name, '-' + str(version) if version else ''
+            )
+            raise MissingPackageException(error)
+
+    ybase.resolveDeps()
+    ybase.buildTransaction()
+
+    try:
+        ybase.processTransaction()
+    except (yum.Errors.YumDownloadError, yum.Errors.YumRPMCheckError) as exc:
+        error = 'Some package dependencies are missing: {0}'.format(exc)
+        raise MissingPackageException(error)
```

Comment:
Suggest you use one of the default salt exception so that salt can "pretty" print the error (no Traceback)
https://docs.saltstack.com/en/latest/ref/internals/salt.exceptions.html 
(Likely `CommandExectuionError`)

Reply:
Ok, I wasn't aware of that, will change.

Reply:
I tried, but I don't see any difference.
There is still an ugly traceback:
```
[ERROR   ] Unhandled exception occurred in function "run: No candidate found for package: toto
[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/state.py", line 1919, in call
    **cdata['kwargs'])
  File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1918, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/utils/decorators/__init__.py", line 558, in _decorate
    return self._call_function(kwargs)
  File "/usr/lib/python2.7/site-packages/salt/utils/decorators/__init__.py", line 263, in _call_function
    raise error
CommandExecutionError: No candidate found for package: toto

[states output]

----------
          ID: Check packages availability
    Function: module.wait
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/site-packages/salt/state.py", line 1919, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1918, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/site-packages/salt/utils/decorators/__init__.py", line 558, in _decorate
                  return self._call_function(kwargs)
                File "/usr/lib/python2.7/site-packages/salt/utils/decorators/__init__.py", line 263, in _call_function
                  raise error
              CommandExecutionError: No candidate found for package: toto
     Started: 07:50:10.257166
    Duration: 5377.859 ms
     Changes:
```

Reply:
Oh yes right seems it's not "pretty" when it's in a state but only when you call the module directly. Nevermind still better to use the predefined salt exceptions :smiley:

Reply:
As discussed on slack, module.run & module.wait do not pretty print exception.
It's only done when the module is called on CLI.